scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -187,14 +186,20 @@ class UDPPeerGroup[M](val config: Config)(
       val nettyChannel: NioDatagramChannel,
       localAddress: InetSocketAddress,
       remoteAddress: InetSocketAddress,
-      val messageSubject: ConnectableSubject[ChannelEvent[M]],
+      val messageQueue: ConcurrentQueue[Task, ChannelEvent[M]],
```

Comment:
I think this should be a private field, so pushing to it is encapsulated to the `ChannelImpl` itself. At least I was looking up and down for where the messages are published, which could be anywhere as the `messageSubject` was accessed directly.